grim.go

```diff
@@ -189,6 +190,26 @@ func writeHookEvent(resultPath string, hook hookEvent) error {
 }
 
 func onHook(configRoot string, config *effectiveConfig, hook hookEvent, logger *log.Logger, action hookAction) error {
+	build := make(chan bool, 1)
+
+	var buildResult error
```

Comment:
You are mixing and matching some things here that you probably shouldn't.  First you are passing that build channel around when you largely don't need to, it could exist only in this function and could be called at the end of the go func.  Second, you are sharing across thread (and closure) boundaries the buildResult variable.  That is bad practice, though in this particular case I don't think it matters.

I would refactor this to use a channel that exists only in this function, and passes the build result error through it or closes the channel on no error.  Then in the select you can watch for error or channel close.  It keeps all the threading code in this function instead of polluting hte other one and it solves the data sharing issue.